settings.php

```diff
@@ -67,6 +83,35 @@
        get_string('send_response_choices', 'logstore_xapi'),
        get_string('send_response_choices_desc', 'logstore_xapi'), 0));
 
+    // Control sending notifications.
+    $settings->add(new admin_setting_configcheckbox('logstore_xapi/enablesendingnotifications',
+        get_string('enablesendingnotifications', 'logstore_xapi'),
+        get_string('enablesendingnotifications_desc', 'logstore_xapi'), 1));
+
+    // Set threshold for sending notifications.
+    $settings->add(new admin_setting_configtext('logstore_xapi/errornotificationtrigger',
+        get_string('errornotificationtrigger', 'logstore_xapi'),
+        get_string('errornotificationtrigger_desc', 'logstore_xapi'), 10, PARAM_INT));
+
+    // Cohorts.
+    $settings->add(new admin_setting_heading('cohorts',
+        get_string('cohorts', 'logstore_xapi'),
+        get_string('cohorts_help', 'logstore_xapi')));
+
+    $cohorts = logstore_xapi_get_cohorts();
+    $arrcohorts = array();
+    foreach ($cohorts as $cohort) {
+        $arrcohorts[$cohort->id] = $cohort->name;
+    }
+
+    $settings->add(new admin_setting_configmulticheckbox('logstore_xapi/cohorts',
+        get_string('includecohorts', 'logstore_xapi'), '', '', $arrcohorts));
+
+    // Additional email addresses.
+    $settings->add(new admin_setting_configtext('logstore_xapi/send_additional_email_addresses',
+        get_string('send_additional_email_addresses', 'logstore_xapi'),
+        get_string('send_additional_email_addresses_desc', 'logstore_xapi'), '', PARAM_TEXT));
+
```

Comment:
Figured it out! It's because I didn't have any cohorts on the Moodle instance. Should probably account for this here.

Reply:
Hi David, I have changed my code to not display this setting if no cohorts are setup on the Moodle instance. This should fix the problem during db upgrades.

Reply:
Sorry David, I think this should be being merged into master-jisc not master. I will discuss this with Gordon in the morning.

Reply:
No worries. It looks like Gordon merged this into master-jisc yesterday. We chatted over email, but always better to check.